indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -149,6 +151,10 @@ public String getClasspathPrefix()
   @Override
   public TaskStatus run(TaskToolbox toolbox) throws Exception
   {
+    //Setting up UsedSegmentLister which is used by DatasourcePathSpec to get the list of used
+    //segments
+    DatasourcePathSpec.setUsedSegmentLister(new OverlordActionBasedUsedSegmentLister(toolbox));
```

Comment:
sounds true, let me move it to HadoopDetermineConfigInnerProcessing.runTask(..) which will happen inside the other classloader .
ugh, this whole classloader thing coupled with supporting different ways of doing things in standalone hadoop indexer vs overlord task is becoming very ugly